cpp/arcticdb/storage/memory_layout.hpp

```diff
@@ -101,15 +101,15 @@ enum class BitmapFormat : uint8_t {
 // pointer to the first block
 struct EncodedField {
     EncodedFieldType type_ = EncodedFieldType::UNKNOWN;
-    uint16_t shapes_count_ = 0u;
-    uint16_t values_count_ = 0u;
+    uint32_t shapes_count_ = 0u;
+    uint32_t values_count_ = 0u;
```

Comment:
Could add some checks that what we are decoding will fit in the `uint32_t` so we fail in a more obvious way?

Reply:
Yeh, one to discuss with @willdealtry when he's back I think, as the same logic applies to any of these ints that are <64 bits wide

Reply:
How did we end up storing more than 65k blocks?